main/src/cgeo/geocaching/apps/cache/WhereYouGoApp.java

```diff
@@ -20,7 +20,7 @@
 import org.apache.commons.lang3.StringUtils;
 
 public class WhereYouGoApp extends AbstractGeneralApp {
-    private static final Pattern PATTERN_CARTRIDGE = Pattern.compile("(" + Pattern.quote("http://www.wherigo.com/cartridge/details.aspx?") + ".*?)" + Pattern.quote("\""));
+    private static final Pattern PATTERN_CARTRIDGE = Pattern.compile("(https?\\:\\/\\/www\\.wherigo\\.com\\/cartridge\\/details\\.aspx\\?[Cc][Gg][Uu][Ii][Dd]=[-0-9a-zA-Z]*)");
```

Comment:
Are the double "\" really needed for escaping?
Please start with "a href=\"" in the pattern to find the right position. Then it is clear that the next "\">" will end the match.
Maybe you could refactor the code to move the pattern to the other in `cgeo.geocaching.connector.gc.GCConstants`

Reply:
> Are the double "\" really needed for escaping?

Find by myself that this is needed. With `Pattern.quote` is the code better readable.

Reply:
The tests in WhereYouGoAppTest should also be updated.

Reply:
> > Are the double "" really needed for escaping?
> 
> Find by myself that this is needed. With `Pattern.quote` is the code better readable.

I had tried it with `Pattern.quote` before, as the original pattern also used this, but the mix of unquoted and quoted regexp did not work in this case, so I switched to direct quoting in this case.

Reply:
> Please start with "a href=\"" in the pattern to find the right position. Then it is clear that the next "\">" will end the match.

I would rather stay without a `a href` part, as the link does not necessarily have to be enclosed in an `a` tag.

> Maybe you could refactor the code to move the pattern to the other in `cgeo.geocaching.connector.gc.GCConstants`

Is that the right place? WIG is from geocaching.com as well, that's true, but it is not related to the GC connector, so I think it should stay separate.

Reply:
> The tests in WhereYouGoAppTest should also be updated.

added two tests to catch both new cases - https and URL as `a href` description (or anywhere else in the description)

Reply:
> I would rather stay without a `a href` part, as the link does not necessarily have to be enclosed in an `a` tag.

You are right, we are parsing "freetext".

Reply:
> Is that the right place? WIG is from geocaching.com as well, that's true, but it is not related to the GC connector, so I think it should stay separate.

Mmm, yes. If we make a generalized pattern file, then the place should also be central. Let´s as is.